### ISSUE TITLE
[java] ReturnEmptyCollectionRatherThanNull - false positive with stream and lambda

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#3721]
**Affects PMD Version:** 6.42.0-SNAPSHOT

**Rule:** [ReturnEmptyCollectionRatherThanNull](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#returnemptycollectionratherthannull)

**Description:** the returned type is actually `Future<String>` and not a collection - although the outer method returns a collection (Map).

**Code Sample demonstrating the issue:**

```java
    @SuppressWarnings("PMD.ReturnEmptyCollectionRatherThanNull")
    private Map<Path, List<String>> joinFutures(Map<Path, List<Future<String>>> map) {
        Map<Path, List<String>> joined = new HashMap<>();

        for (Path p : map.keySet()) {

            List<String> evaluatedResult = map.get(p).stream()
                                              .map(f -> {
                                                  try {
                                                      return f.get();
                                                  } catch (InterruptedException | ExecutionException e) {
                                                      e.printStackTrace();
                                                      return null; // <----- false positive here
                                                  }
                                              })
                                              .filter(Objects::nonNull)
                                              .sorted(Comparator.naturalOrder())
                                              .collect(Collectors.toList());

            if (!evaluatedResult.isEmpty()) {
                joined.put(p, evaluatedResult);
            }
        }
        return joined;
    }
```

https://github.com/pmd/pmd/blob/f38e99c65ed860958001c0435f7111400b40dbd1/pmd-doc/src/main/java/net/sourceforge/pmd/docs/DeadLinksChecker.java#L235

**Expected outcome:**

PMD reports a violation, but that's wrong. That's a false positive.

**Running PMD through:** Maven
